### ISSUE TITLE
Hide certain chars while typing

### ISSUE THREAD

**Issue by ar7casper** [s-yadav/react-number-format#596]
#### Describe the issue and the actual behavior
Imagine I got an SSN field, and while the user types, I need to substitute the entered number with an asterisk.  Except for the last 4 digits, which we would like to show.
Currently, working with the format prop, I couldn't make it work. 
It seems like format gets the `val` from the input, which I have to work with. 
If a user enters `4` and I replace it with `*`, I lose the data, and it doesn't type any other digits. 
Storing the value in a local state also didn't work. 

```typescript
<NumberFormat format={(val) => {
  if (val.length >= 9) {
    return val.slice(0,9)
    }
    return val;
}}
```

The above snippet works excellent. 
The second I'm changing to - 

```typescript
<NumberFormat format={(val) => {
  if (val.length >= 9) {
    return val.slice(0,9)
    }
  const masked = hideSsn(val);
  return masked;
}}
```

I always get the initial `*` back.
Not sure if you support such a need.

**Comment by jbean96** [comment]
I am looking for this exact use case, would be incredibly helpful if `react-number-format` could support this.

**Comment by AndrewOttavianoAbb** [comment]
+1 to this

**Comment by AndrewOttavianoAbb** [comment]
Agreed 100%! We have a use case where this would be super helpful as well. We thought this would be supported in version 5 because of the [Persian numerals example](https://s-yadav.github.io/react-number-format/docs/customization#card-expiry-field), but I'm not able to get it to work. I either experience the same problem as mentioned above or the component seems to eat my input.

@s-yadav, could this be supported? If it already is, could you please provide an example in the documentation for us to follow? Thank you so much! This is a really helpful, useful, and cool project! We appreciate your work on it!
